model/riscv_sys_regs.sail

```diff
@@ -852,24 +852,31 @@ bitfield Satp64 : bits(64) = {
   PPN  : 43 .. 0
 }
 
-function legalize_satp64(a : Architecture, o : bits(64), v : bits(64)) -> bits(64) = {
-  let s = Mk_Satp64(v);
-  match satp64Mode_of_bits(a, s[Mode]) {
-    None()     => o,
-    Some(Sv32) => o,  /* Sv32 is unsupported for now */
-    Some(_)    => s.bits
-  }
-}
-
 bitfield Satp32 : bits(32) = {
   Mode : 31,
   Asid : 30 .. 22,
   PPN  : 21 .. 0
 }
 
-function legalize_satp32(a : Architecture, o : bits(32), v : bits(32)) -> bits(32) = {
-  /* all 32-bit satp modes are valid */
-  v
+// TODO: This currently assumes all ASID bits are implemented
+// and does not legalize the asid field if asidlen is not the maximum.
+function legalize_satp(
+  a : Architecture,
+  o : xlenbits,
```

Comment:
Suggest adding comment:  /* previous value of satp */

Reply:
Good point - I updated the variable names to be more descriptive rather than adding comments. We should probably do that for all the other `legalize_` functions which also use incomprehensible single letter variables.